xarray/tests/test_dask.py

```diff
@@ -110,8 +110,38 @@ def test_indexing(self):
         self.assertLazyAndIdentical(u[0], v[0])
         self.assertLazyAndIdentical(u[:1], v[:1])
         self.assertLazyAndIdentical(u[[0, 1], [0, 1, 2]], v[[0, 1], [0, 1, 2]])
-        with pytest.raises(TypeError, match=r"stored in a dask array"):
-            v[:1] = 0
+        if LooseVersion(dask.__version__) >= LooseVersion("2021.04.0+17"):
+            arr = Variable(("x"), da.array([1, 2, 3, 4]))
+            expected = Variable(("x"), da.array([99, 2, 3, 4]))
+            arr[0] = 99  # Indexing by integers
+            assert_identical(arr, expected)
+            arr = Variable(("x"), da.array([1, 2, 3, 4]))
+            expected = Variable(("x"), da.array([99, 99, 99, 4]))
+            arr[2::-1] = 99  # Indexing by slices
+            assert_identical(arr, expected)
+            arr = Variable(("x"), da.array([1, 2, 3, 4]))
+            expected = Variable(("x"), da.array([99, 99, 3, 99]))
+            arr[[0, -1, 1]] = 99  # Indexing by a list of integers
+            assert_identical(arr, expected)
+            arr = Variable(("x"), da.array([1, 2, 3, 4]))
+            expected = Variable(("x"), da.array([99, 99, 99, 4]))
+            arr[np.arange(3)] = 99  # Indexing by a 1-d numpy array of integers
+            assert_identical(arr, expected)
+            arr = Variable(("x"), da.array([1, 2, 3, 4]))
+            expected = Variable(("x"), da.array([1, 99, 99, 99]))
+            arr[[False, True, True, True]] = 99  # Indexing by a list of booleans
+            assert_identical(arr, expected)
+            arr = Variable(("x"), da.array([1, 2, 3, 4]))
+            expected = Variable(("x"), da.array([1, 99, 99, 99]))
+            arr[np.arange(4) > 0] = 99  # Indexing by a 1-d numpy array of booleans
+            assert_identical(arr, expected)
+            arr = Variable(("x"), da.array([1, 2, 3, 4]))
+            expected = Variable(("x"), da.array([99, 99, 99, 99]))
+            arr[arr > 0] = 99  # Indexing by one broadcastable Array of booleans
+            assert_identical(arr, expected)
+        else:
+            with pytest.raises(TypeError, match=r"stored in a dask array"):
+                v[:1] = 0
```

Comment:
Nitpick: I was wondering if we could use `@pytest.mark.parametrize` here to parametrize this test so as to avoid code duplication?

Reply:
I definitely agree, but I think I'd prioritize merging quickly. For a first PR, it's great code overall. We could add a `TODO` to the current code so it doesn't look like best practice to the next person.

Reply:
I concur...